loadtest.go

```diff
@@ -67,28 +67,43 @@ func (l *loadTest) runUser() {
 		cancel()
 	}()
 
-	// initialise a client, reporting any errors that occur
-	l.log.Debug("initialising client")
-	client, err := l.newClientFunc(ctx, l.w.Conf(), l.log)
-	if err != nil {
-		l.log.Debug("error initialising client", "err", err)
-		l.w.boomer.RecordFailure("ablyboomer", "client", 0, err.Error())
-		return
-	}
-	defer client.Close()
+	for {
+		if ctx.Err() != nil {
+			break
+		}
 
-	// run the enabled tasks until the loadTest is stopped
-	errG, ctx := errgroup.WithContext(ctx)
-	if l.w.Conf().Subscriber.Enabled {
-		errG.Go(func() error { return l.runSubscriber(ctx, client, userNum) })
-	}
-	if l.w.Conf().Publisher.Enabled {
-		errG.Go(func() error { return l.runPublisher(ctx, client, userNum) })
-	}
-	if l.w.Conf().Presence.Enabled {
-		errG.Go(func() error { return l.runPresence(ctx, client, userNum) })
+		runCtx := ctx
+		runCancel := func() {}
+		if reconnectInterval := l.w.conf.Subscriber.ReconnectInterval; reconnectInterval > 0 {
```

Comment:
I wonder if we should now call this something like `conf.UserLifetime`, since it's no longer just about subscriber reconnects (e.g. it could now be used to control the rate at which users enter and leave presence, too), and more about how long the user should perform its tasks for before starting again

Reply:
yes I only realised the name was now wrong after I set the configuration